File: SNGINT.py
def main():
	for i in xrange(input()):
		n = input()

		if not n:
			print "10"
			continue

		if n<10:
			print n
			continue

		stri = []

		for i in xrange(9,1,-1):
			if n==1:
				break

			while not n%i:
				n/=i
				stri.append(i)

		k = ""

		for i in xrange(len(stri)-1,-1,-1):
			k+=str(stri[i])

		if n!=1:
			print "-1"
		else:
			print k

main()

File: SNGINT.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long int ll;

main()
{
	freopen("input.txt","r",stdin);
	freopen("output.txt","w",stdout);

	ll t;
	scanf("%lld",&t);

	while(t--)
	{
		ll n;
		scanf("%lld",&n);

		if (n==1)
		{
			printf("-1\n");
			continue;
		}
		if (not n)
		{
			printf("10\n");
			continue;
		}

		ll c[100],k=0;
		for(ll i=9;i>=2;i--)
			while(n%i==0)
			{
				n/=i;
				c[k++]=i;
			}

		for(ll i=k-1;i>=0;i--)
			printf("%lld",c[i]);

		if (n!=1)
			printf("-1");

		printf("\n");

	}
}
